autometa/common/external/diamond.py

```diff
@@ -28,204 +28,359 @@
+        """
+        Operator overloading to return the string representation of the class objects
+
+        Returns
+        -------
+        str
+            String representation of query sequence ID, followed by total number of hits and finally the highest
+            bit score of all the hits
+        """
         return f"{self.qseqid}; {len(self.sseqids)} sseqids; top hit by bitscore: {self.get_top_hit()}"
 
     def __eq__(self, other_hit):
+        """
+        Operator overloading to compare two objects of the DiamondResult class
+
+        Parameters
+        ----------
+        other_hit : DiamondResult object 
+            Other DiamondResult object to compare with
+
+        Returns
+        -------
+        Boolean
+            True if qseqid corresponding to both DiamondResult objects are equal, else False
+        """
         if self.qseqid == other_hit.qseqid:
             return True
         else:
             return False
 
     def __add__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
+        """
+        Operator overloading to update (add) the sseqids dictionary with the other sseqid hit dictionary.
+        The addition will only be successful if both the DiamondResult objects have the same qseqid
+
+        Parameters
+        ----------
+        other_hit : DiamondResult object
+            Other DiamondResult object to add 
+
+        Returns
+        -------
+        DiamondResult object
+            Another sseqid dict as now been added (updated) to the sseqids dict in the object
+
+        Raises
+        ------
+        AssertionError
+            Query sequences are not equal
+        """
+        assert (
+            self == other_hit
+        ), f"qseqids do not match! {self.qseqid} & {other_hit.qseqid}"
         self.sseqids.update(other_hit.sseqids)
         return self
 
     def __sub__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
-        self.sseqids.pop(other_hit.sseqid)
+        """
+        Operator overloading to remove (subtract) the other sseqid hits dictionary from the sseqids dictionary.
+        The subtraction will only be successful if both the DiamondResult objects have the same qseqid
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            Other DiamondResult object to subtract
+
+        Returns
+        -------
+        DiamondResult object
+            A sseqid dict as now been subtracted (removed) from the sseqids dict in the object
+        
+        Raises
+        ------
+        AssertionError
+            Query sequences are not equal
+        """
+        assert (
+            self == other_hit
+        ), f"qseqids do not match! {self.qseqid} & {other_hit.qseqid}"
+        for key in list(other_hit.sseqids.keys()):
+            try:
+                self.sseqids.pop(key)
+            except KeyError as err:
+                raise err
```

Comment:
A few problems here.

1. There is no need to use the `list` function when also employing the `dict.keys` method. When iterating through a dictionary, the default behavior is to iterate through the keys. So the method and the function are unnecessary.
2. It would be more helpful to be descriptive with your variables.  i.e. `sseqid` rather than `key`.
2. The error message is not helpful and this is 3 extra lines to give the same exact message as just brute forcing `self.sseqids.pop(key)`
3. I think the `try` and `except` block could be useful here, but you need to provide more helpful information.

```suggestion
        for sseqid in other_hit.sseqids:
            try:
                self.sseqids.pop(sseqid)
            except KeyError:
                raise KeyError(<'Some useful error message here'>)
```